programs/util.c

```diff
@@ -87,6 +94,239 @@ U32 UTIL_isDirectory(const char* infilename)
     return 0;
 }
 
+int UTIL_createDir(const char* outDirName)
+{
+    int r;
+    if (UTIL_isDirectory(outDirName))
+        return 0;   /* no need to create if directory already exists */
+
+#if defined(_MSC_VER) || defined(__MINGW32__) || defined (__MSVCRT__)
+    r = _mkdir(outDirName);
+    if (r || !UTIL_isDirectory(outDirName)) return 1;
+#else
+    r = mkdir(outDirName, S_IRWXU | S_IRWXG | S_IRWXO); /* dir has all permissions */
+    if (r || !UTIL_isDirectory(outDirName)) {
+        errno = 0;
+        return 1;
+    }
+#endif
+    errno = 0;
+    return 0;   /* success */
+}
+
+/* function requires posix >= 2001 */
+int UTIL_getRealPath(const char* relativePath, char* absolutePath) {
+#if defined(_MSC_VER) || defined (__MINGW32__) || defined (__MSVCRT__)
+    if (_fullpath(absolutePath, relativePath, LIST_SIZE_INCREASE) == NULL)
+        return 1;
+#elif (PLATFORM_POSIX_VERSION >= 200112L)
+#define _BSD_SOURCE
+    char* r;
+    r = realpath(relativePath, absolutePath);
+    if (r == NULL && errno != ENOENT /* we will be using this to generate "FAKE" paths, so ignore this err */) {
+        errno = 0;
+        return 1;
+    }
+#else
+    UTIL_DISPLAYLEVEL(1, "Unable to process realpath due to platform limitations\n", relativePath, absolutePath);
+    return -1;  /* realpath() not supported */
+#endif
+    errno = 0;
+    return 0;
+}
+
+int UTIL_createPath(const char* path)
+{
+    char* ptr;
+    char c;
+    int result;
+
+    c = '/';
+    #if defined(_MSC_VER) || defined(__MINGW32__) || defined (__MSVCRT__)   /* windows support */
+    c = '\\';
+    #endif
+    result = 0;
+
+    /* approach is to start from the first folder in path, and iteratively try to construct a dir at each '/' */
+    for (ptr = strchr(path+1, c); ptr; ptr = strchr(ptr+1, c)) {
+        *ptr = '\0';
+        result = UTIL_createDir((const char*) path);
+        if (result) {
+            UTIL_DISPLAYLEVEL(9, "Unsuccessful directory creation: either already exists or system error\n");
```

Comment:
This is inaccurate, I think? `UTIL_createDir()` succeeds when the directory already exists.

Reply:
Totally - was thinking of mkdir.